model/testing/src/icon4py/model/testing/pytest_config.py

```diff
@@ -123,6 +123,9 @@ def pytest_addoption(parser):
 
 
 def pytest_runtest_setup(item):
+    for marker in item.own_markers:
+        if marker.name.startswith("embedded") and item.config.getoption("--backend") == "embedded":
+            pytest.skip("test not compatible with embedded backend")
```

Comment:
This is for all non-StencilTests, right? Then we should probably use the same function as in helpers.py.